### PR TITLE
add title-dropdown (fixes #92)

### LINKED ISSUE #92: Improve complex field lists interface
The interface for complex list fields (such as references) is currently confusing due to lists nested inside of objects.

For example, in the image below there are two X buttons to delete an element from the array and the whole `misc` field.

image

In order to declutter the interface and as a bonus support https://github.com/inveniosoftware-contrib/ng2-json-editor/issues/7 I suggest to modify the complex list fields UI removing the add and delete buttons:

image

And instead making the label a [button-dropdown](http://getbootstrap.com/components/#btn-dropdowns):

editor

### PR DESCRIPTION
**Pull request by harunurhan**
* Adds title-dropdown that is used to hide some action buttons
such as add-new-element and delete.

* Removes extra space from button-holder column.

### CODE REVIEW

File: src/complex-list-field/complex-list-field.component.html [hunk 1]
```diff
@@ -42,8 +42,14 @@
         <tr *ngFor="let key of keys[i] | addAlwaysShowFields:schema.items | filterAndSortBySchema:schema.items; trackBy:trackByFunction">
           <td class="label-holder">
             <div>
-              <label>{{key | underscoreToSpace}}</label>
-              <add-new-element-button *ngIf="schema.items.properties[key].type === 'array'" [path]="getValuePath(pIndex, key)" [schema]="schema.items.properties[key]"></add-new-element-button>
+              <title-dropdown [title]="key | underscoreToSpace">
```
Comment (jmartinm) [inline src/complex-list-field/complex-list-field.component.html]: Can you make the style of this dropdown like the bootstrap one? (See that the whole row is clickable and has a background on hover)

dropdown

File: src/complex-list-field/complex-list-field.component.ts [hunk 2]
```diff
@@ -100,10 +100,19 @@ export class ComplexListFieldComponent extends AbstractListFieldComponent implem
     }
   }
 
-  onFieldAdd(field: string, index: number) {
+  onFieldAdd(index: number, field: string) {
     this.keys[index].push(field);
   }
 
+  deleteField(index: number, field: string) {
+    // remove it the field from the element of list (record)
```
Comment (jmartinm) [inline src/complex-list-field/complex-list-field.component.ts]: This comment is a bit difficult to understand as is, might be better without it.

File: src/complex-list-field/complex-list-field.component.ts [hunk 3]
```diff
@@ -100,10 +100,19 @@ export class ComplexListFieldComponent extends AbstractListFieldComponent implem
...
+    this.values = this.values.removeIn([index, field]);
+    this.jsonStoreService.setIn(this.path, this.values);
+
+    // remove it from keys too, so that it will not be displayed as empty
```
Comment (jmartinm) [inline src/complex-list-field/complex-list-field.component.ts]: This comment seems fine.

File: src/title-dropdown/title-dropdown.component.scss [hunk 4]
```diff
@@ -0,0 +1,8 @@
+.title-dropdown-toggle {
```
Comment (jmartinm) [inline src/title-dropdown/title-dropdown.component.scss]: See comment above about styling.

File: src/complex-list-field/complex-list-field.component.html (hunk 1 above)
Comment (harunurhan) [inline src/complex-list-field/complex-list-field.component.html]: Actually this was my intention, and I had this on hover background but when I moved the html into `title-dropdown`, somehow it disappeared. However I'll look into it again.

**Comment by harunurhan** [comment]
@jmartinm addressed all comments.

**Pull request merged**